### DISCUSSION TITLE
is https://www.prisma.io/docs/guides/performance-and-optimization/connection-management up to date?

### DISCUSSION THREAD

**Discussion by capaj** [prisma/prisma#18156] [Q&A] [answered]
https://www.prisma.io/docs/guides/performance-and-optimization/connection-management docs state that:

> default pool size (num_physical_cpus * 2 + 1)

but on my CI machine with 32 cores I am seeing connection limit of 33 without explicitly setting a connection limit. Are docs out of date?
![image](https://user-images.githubusercontent.com/1305378/222130833-a6ba094b-a58e-4bc1-a66f-4fbaeede25bc.png)

**Comment by janpio** [comment]
That would probably more point to use not being to pick those up correctly.

Can you share a bit more about this machine?

**Reply by janpio** [reply, marked as answer]
Here is the code that leads to this default value:
```
num_cpus::get_physical() * 2 + 1
```
https://github.com/prisma/quaint/blob/27cf871d0af0c59c9931eb2950de10eea433cd3c/src/pooled.rs#L190-L192

The underlying function is https://docs.rs/num_cpus/latest/num_cpus/fn.get_physical.html
> Returns the number of physical cores of the current system.

**Reply by nikolasburk** [reply]
Hey @capaj  👋 did Jan's answer clear this up for you?

Do you have any more questions around this issue? If no, it would be great if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
